modules/product/src/main/java/com/opengamma/strata/product/GenericSecurityPosition.java

```diff
@@ -0,0 +1,663 @@
+   * 
+   * @param security  the underlying security
+   * @param netQuantity  the net quantity of the underlying security
+   * @return the position
+   */
+  public static GenericSecurityPosition ofNet(GenericSecurity security, long netQuantity) {
+    return ofNet(PositionInfo.empty(), security, netQuantity);
+  }
+
+  /**
+   * Obtains an instance from position information, security and net quantity.
+   * <p>
+   * The net quantity is the long quantity minus the short quantity, which may be negative.
+   * 
+   * @param positionInfo  the position information
+   * @param security  the underlying security
+   * @param netQuantity  the net quantity of the underlying security
+   * @return the position
+   */
+  public static GenericSecurityPosition ofNet(PositionInfo positionInfo, GenericSecurity security, long netQuantity) {
+    long longQuantity = netQuantity >= 0 ? netQuantity : 0;
+    long shortQuantity = netQuantity >= 0 ? 0 : Math.negateExact(netQuantity);
+    return new GenericSecurityPosition(positionInfo, security, longQuantity, shortQuantity);
+  }
+
+  /**
+   * Obtains an instance from the security, long quantity and short quantity.
+   * <p>
+   * The long quantity and short quantity must be zero or positive, not negative.
+   * 
+   * @param security  the underlying security
+   * @param longQuantity  the long quantity of the underlying security
+   * @param shortQuantity  the short quantity of the underlying security
+   * @return the position
+   */
+  public static GenericSecurityPosition ofLongShort(GenericSecurity security, long longQuantity, long shortQuantity) {
+    return ofLongShort(PositionInfo.empty(), security, longQuantity, shortQuantity);
+  }
+
+  /**
+   * Obtains an instance from position information, security, long quantity and short quantity.
+   * 
+   * @param positionInfo  the position information
+   * @param security  the underlying security
+   * @param longQuantity  the long quantity of the underlying security
+   * @param shortQuantity  the short quantity of the underlying security
+   * @return the position
+   */
+  public static GenericSecurityPosition ofLongShort(
+      PositionInfo positionInfo,
+      GenericSecurity security,
+      long longQuantity,
+      long shortQuantity) {
+
+    return new GenericSecurityPosition(positionInfo, security, longQuantity, shortQuantity);
+  }
+
+  @ImmutableDefaults
+  private static void applyDefaults(Builder builder) {
+    builder.info = PositionInfo.empty();
+  }
+
+  //-------------------------------------------------------------------------
+  /**
+   * Gets the security identifier.
+   * <p>
+   * This identifier uniquely identifies the security within the system.
+   * 
+   * @return the security identifier
+   */
+  @Override
+  public SecurityId getSecurityId() {
+    return security.getSecurityId();
+  }
+
+  /**
+   * Gets the currency of the trade.
+   * <p>
+   * This is typically the same as the currency of the product.
```

Comment:
"typically"? Is it ever a different currency?

Also it should probably say "security" instead of "product".